### PR TITLE
feat(queue_pickle.py): can now save and load cloud jobs in progress

### PR DESCRIPTION
**Pull request by william-silversmith**
New script. Save to and load cloud queues from a file. This can be useful if a big job gets interrupted and other high priority jobs are needed.

In this case, the pinky100 job was interrupted, but it was taking a week so I didn't want to restart it. Instead, I can now save it and restart it at a more convenient time.

### CODE REVIEW

File: python/neuroglancer/pipeline/scripts/queue_pickle.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  click.echo("Remember to ensure the queue is not running and no leases are active!")
+
+  queue = []
+
+  if file is None:
+    timestamp = datetime.today().strftime("%Y-%m-%d-%H:%M")
+    file = './{}-{}.json'.format(queue_name, timestamp)
+  else:
+    queue = deserialize(file)
+    print "Restarting with {} as seed. Tasks: {}".format(file, len(queue))
```
Comment (wongwill86) [inline python/neuroglancer/pipeline/scripts/queue_pickle.py]: wait there isn't any restarting here right? We're only saving. so wouldn't the message be more appropriate as something along the lines of ```Using existing seed file {} .... ```

Comment (william-silversmith) [inline python/neuroglancer/pipeline/scripts/queue_pickle.py]: When I was making it it sometimes failed until I figured it out. It's not the most solid script in the world. The idea is that maybe it crashes and you can keep going without losing your work....

**Comment by william-silversmith** [comment]
This is really designed exclusively for google cloud pull queues. It could also be useful in the newer queue, but I don't really know enough about it to say if this is the best way to do it. Certainly, since it conforms to the google interface this approach might just work anyway.

**Comment by wongwill86** [comment]
Just talked to @tartavull, this won't work with the new dependency task queue from #153.
FYI as of #153, the dependency task queue is now the [default task queue](https://github.com/seung-lab/neuroglancer/pull/153/files#diff-3c7a7f91ebc9a8a8d5e5cda718e6d519R102)
 Looks like it's using the same the same queue interface so it's not clear that this script doesn't support it.  We should either throw when trying to run on a dependency queue or support it using the [list](https://github.com/seung-lab/neuroglancer/pull/153/files#diff-1530c448f340ba195076b77ceda4f925R75) function.

**Comment by william-silversmith** [comment]
Hmm, I haven't been able to use the new default task queue due to a few issues so I'm still using the Google Task Queue for now. Would it be alright to merge this?

File: python/neuroglancer/pipeline/task_queue.py [hunk 2]
```diff
@@ -212,22 +221,30 @@ def update(self, task):
...
             groupByTag=(tag is not None),
             tag=tag,
         ).execute(num_retries=6)
 
+    def lease(self, tag=None):
+        """
+        Acquires a lease on the topmost N unowned tasks in the specified queue.
+        Required query parameters: leaseSecs, numTasks
+        """
+        tasks = self.raw_lease(numTasks=1, leaseSecs=600, tag=tag)
```
[the remark this answers is not in the document]
Comment (william-silversmith) [reply python/neuroglancer/pipeline/task_queue.py]: Removed default arguments.

**Comment by william-silversmith** [comment]
We recently used this to reload the pinky100_v0 download queue.

**Comment by tartavull** [comment]
@william-silversmith What's the status on this?

**Comment by william-silversmith** [comment]
I think it would be good to merge as-is. It's designed to work with the google pull queue. We can improve it later.

**Comment by william-silversmith** [comment]
I'm going to migrate this to igneous.

**Pull request closed without merging**